src/HttpClientFactory/Http/src/DependencyInjection/HttpClientBuilderExtensions.cs

```diff
@@ -514,5 +524,35 @@ public static IHttpClientBuilder SetHandlerLifetime(this IHttpClientBuilder buil
             builder.Services.Configure<HttpClientFactoryOptions>(builder.Name, options => options.HandlerLifetime = handlerLifetime);
             return builder;
         }
+
+        // See comments on HttpClientMappingRegistry.
+        private static void ReserveClient(IHttpClientBuilder builder, Type type, string name)
+        {
+            var registry = (HttpClientMappingRegistry)builder.Services.Single(sd => sd.ServiceType == typeof(HttpClientMappingRegistry)).ImplementationInstance;
```

Comment:
@davidfowl if you don't like this I have another cool idea https://github.com/aspnet/Extensions/commit/b44b4590453592d4df9de449e22f36f78c028f4a

Reply:
This is horrible but it's less horrible than the other way.

Reply:
I really wish you had dont the `cool idea` way - just hit a bug with this in Orchard Core. The Cool Way would have worked :(

Reply:
I was trying to get @davidfowl to allow `ConditionalWeakTable` into the codebase for **years** 😆